parsec/core/gui/workspaces_widget.py

```diff
@@ -241,6 +243,21 @@ def __init__(self, core, jobs_ctx, event_bus, **kwargs):
         self.sharing_updated_qt.connect(self._on_sharing_updated_qt)
         self._workspace_created_qt.connect(self._on_workspace_created_qt)
 
+        self.filter_user_info = None
+        self.shared_button.hide()
+
+    def remove_user_filter(self):
+        self.filter_user_info = None
+        self.shared_button.hide()
+        self.reset()
+
+    def set_user_info(self, user_info):
+        self.filter_user_info = user_info
+        self.shared_button.show()
+        self.shared_button.setText(
+            _("TEXT_WORKSPACE_FILTERED_user").format(user=user_info.human_handle.label)
```

Comment:
Should use `user_info.short_user_display` here. Some users may not have a `human_handle`.